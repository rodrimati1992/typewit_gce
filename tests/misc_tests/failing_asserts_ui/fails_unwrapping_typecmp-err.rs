#[track_caller]
const fn sneaky() -> usize {
    std::panic::Location::caller().column() as usize
}


const fn foo<T>() {
    _ = typewit_gce::gce_int_eq!(sneaky(), sneaky());
}


fn main() {
    foo::<()>();
}


