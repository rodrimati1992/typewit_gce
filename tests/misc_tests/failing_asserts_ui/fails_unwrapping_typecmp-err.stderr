error[E0080]: evaluation panicked:
              `typewit_gce::gce_int_eq` was passed a const expression whose value can be different in different uses

  --> src/./root_module_hidden_items.rs
   |
   |             let equal_consts = type_cmp.expect_eq(err_msg);
   |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^ evaluation of `typewit_gce::_::<impl typewit_gce::__GceIntEqHelper<usize, typewit_gce::const_marker::Usize<34>, typewit_gce::const_marker::Usize<44>>>::NEW` failed here

note: erroneous constant encountered
  --> tests/misc_tests/failing_asserts_ui/fails_unwrapping_typecmp-err.rs:11:9
   |
11 |     _ = typewit_gce::gce_int_eq!(sneaky(), sneaky());
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this note originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)
